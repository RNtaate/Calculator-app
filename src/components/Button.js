import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const { name } = props;

  let handleClick = (e) => {
    props.clickHandler(e.target.value);
  }

  return (
  <button type="button" onClick={handleClick} value={name}>{name}</button>
  );
};

Button.defaultProps = {
  name: '',
};

Button.propTypes = {
  name: PropTypes.string,
};

export default Button;
