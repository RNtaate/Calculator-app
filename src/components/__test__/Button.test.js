import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../Button';

import renderer from 'react-test-renderer';
import {render} from '@testing-library/react';

it('renders Navbar component without crashing', () => {
  const myDiv = document.createElement('div');
  ReactDOM.render(<Button />, myDiv);
});

it('renders the button component correctly using props', () => {
  const{getByRole} = render(<Button name="2"/>);
  expect(getByRole('button')).toHaveTextContent('2');
});

it('renders the button component correctly using default props values', () => {
  const{getByRole} = render(<Button />);
  expect(getByRole('button')).toHaveTextContent('');
});

it('matches the button component snapshot', () => {
  const tree = renderer.create(<Button name="3"/>);
  expect(tree).toMatchSnapshot();
});