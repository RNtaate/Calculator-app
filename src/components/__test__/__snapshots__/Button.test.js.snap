// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the button component snapshot 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "",
      "width": "25%",
    }
  }
  type="button"
  value="3"
>
  3
</button>
`;
